azavea/climate-change-lab
# Update most dependencies

## Overview

Updates to the latest minor version of most dependencies.

### Notes

- Angular, rxjs, and typescript could not be updated due to type conflicts with `climate-change-components`, and will need to be updated as part of, or after, updating `climate-change-components` to the latest version. 
- I wasn't able to run `yarn run lint` but it looks like that's because it's missing a configuration file that I'm not sure where to find (`The specified path does not exist: 'e2e/tsconfig.e2e.json'.`). I'm looking for help resolving that, and I'll update the README with whatever I do. --Edit-- This is also causing CI to fail. I'm not really sure about where to get started with this; it might be necessary to pair if there's not a quick and easy answer.
- Also includes an update to the README, because it looks like the app doesn't use an API key--it just redirects to the Climate Change API to set a cookie.


## Testing Instructions

 * `yarn install`
 * `yarn run X` everything in the `scripts` key of `package.json` and confirm there are no type errors (a few of them failed for reasons that looked unrelated to dependency upgrades to me).
 * Point your `src/app/constants.ts` to `http://localhost:8080` and bring up your Climate Change API VM. Click around in the lab and make sure there are no console errors and everything works. I don't know the Lab interface very well, so it is very possible I missed some functionality that could have been broken by one of these updates; thorough testing here will be appreciated.

## Checklist
- [x] `yarn run lint` clean? <-- I wasn't able to get this to run, see above.

Closes #295
